From 1a5ed68d82c09edb752501f4a8c1b0e6ec9bf45f Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Wed, 20 Mar 2024 17:49:46 +0200
Subject: [PATCH] [SW-156330] hl: ppboot: fixed tpm startup prints

Added error code print in startup failed message.
Additionally, added print to highlight the RC_INITIALIZE
return scenario.

Change-Id: I2bfabbe88c8c2ea2cf4d4e50d741c176624fc345
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/lib/common/tpm2.c b/lib/common/tpm2.c
index 3c6a879..dcdcbb3 100644
--- a/lib/common/tpm2.c
+++ b/lib/common/tpm2.c
@@ -699,12 +699,12 @@
 	 * but will return RC_INITIALIZE otherwise.
 	 */
 	ret = tpm_spi_sendrecv_command(command_v2, NULL, NULL);
-	if (ret && ret != TPM2_RC_INITIALIZE) {
-		preboot_puts("ret != TPM2_RC_INITIALIZE");
-		return ret;
+	if (ret == TPM2_RC_INITIALIZE) {
+		preboot_puts("TPM already initialized\n");
+		return 0;
 	}
 
-	return 0;
+	return ret;
 }
 
 static int tpm2_unpack_byte_string(const u8 *str, size_t size,
diff --git a/lib/common/tpm_auth.c b/lib/common/tpm_auth.c
index eec8578..777c94c 100644
--- a/lib/common/tpm_auth.c
+++ b/lib/common/tpm_auth.c
@@ -63,7 +63,7 @@
 
 	ret = tpm2_startup(TPM2_SU_CLEAR);
 	if (ret) {
-		preboot_puts("TPM: tpm2_startup failed\n");
+		preboot_puts_hex("TPM: tpm2_startup failed. Error: ", ret, false);
 		cpu_boot_err_set(CPU_BOOT_ERR_TPM_FAIL);
 		return ret;
 	}
@@ -90,7 +90,7 @@
 
 	ret = tpm2_startup(TPM2_SU_CLEAR);
 	if (ret) {
-		preboot_puts("TPM: tpm2_startup failed\n");
+		preboot_puts_hex("TPM: tpm2_startup failed. Error: ", ret, false);
 		cpu_boot_err_set(CPU_BOOT_ERR_TPM_FAIL);
 	}
 
